### PR TITLE
Update compact to new sqw/dnd style

### LINKED ISSUE #1023: Outdated sqw interface calls in `compact`
`Horace/horace_core/sqw/@SQWDnDBase/compact.m` fails on new `sqw` and `DnD` data

```
>> x = sqw('w3d_sqw.sqw');
>> y = compact(x);
No public property 'data_' for class ''sqw''.

Error in SQWDnDBase/compact (line 24)
    ndim = length(win(n).data_.p);
 
```
Replacing all instances of `data_` with `data` still results in failure as unable to set `data.p` directly.

This also suggests the function is untested.

### PR DESCRIPTION
**Pull request by oerc0122**
Fix compact for eliminating empty edge-bins 

Fixes #1023

### CODE REVIEW

File: horace_core/sqw/@DnDBase/compact.m [hunk 1]
```diff
@@ -0,0 +1,84 @@
+function dout = compact(din)
+% Squeezes the data range in a dnd object to eliminate empty bins
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@DnDBase/compact.m:2]:
```suggestion
% Squeezes the data range in a dnd object or array of such objects to eliminate empty bins
```

File: horace_core/sqw/@DnDBase/compact.m [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+%
+% Syntax:
+%   >> dout = compact(din)
+%
+% Input:
+% ------
+%   din         Input object
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@DnDBase/compact.m:9]:
```suggestion
%   din         Input object(s)
```

File: horace_core/sqw/@DnDBase/compact.m [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+%
+% Output:
+% -------
+%   dout        Output object, with length of axes reduced to yield the
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@DnDBase/compact.m:13]:
```suggestion
%   dout        Output object(s), with length of axes reduced to yield the
```

File: horace_core/sqw/@DnDBase/compact.m [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+%               smallest cuboid that contains the non-empty bins.
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@DnDBase/compact.m:14]:
```suggestion
%               smallest cuboid for each input object that contains the non-empty bins.
```

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]

**Review by abuts — COMMENTED** [review]
This blows my mind. I do not understsand if it works or not. e.g. len(din(n).p) can not be zero. p is derived property of `nbin_all_dims`.
Need at least some unit tests.

As small note. I would add abstract compact to `SQWDnDBase` to have common interface for `sqw` and `DnD` objects -- further relocation is good

File: horace_core/sqw/@sqw/compact.m [hunk 5]
```diff
@@ -0,0 +1,25 @@
+function wout = compact(win)
+% Squeezes the data range in an sqw or dnd object to eliminate empty bins
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@sqw/compact.m:2]: do you mean empty bins at the extrema of the ranges only? does not sound as if internal empty bins get eliminated.

the declaration agrees with this

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]

File: horace_core/sqw/@sqw/compact.m [hunk 6]
```diff
@@ -0,0 +1,28 @@
+function wout = compact(win)
+% Squeezes the data range in an sqw object to eliminate empty bins on
+% borders
+%
+% Particularly of use for contracting dnds for plotting
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@sqw/compact.m:5]: not sure about this as this function contracts sqws.

File: horace_core/sqw/@sqw/compact.m [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+%
+% Syntax:
+%   >> wout = compact(win)
+%
+% Input:
+% ------
+%   win         Input object
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@sqw/compact.m:12]:
```suggestion
%   win         Input sqw object
```

File: horace_core/sqw/@DnDBase/compact.m [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+%
+% If an array of DnDs is provided, this will return a cell array
+% otherwise it will return a DnD scalar
+%
+% Syntax:
+%   >> dout = compact(din)
+%
+% Input:
+% ------
+%   din         Input object(s)
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@DnDBase/compact.m:15]:
```suggestion
%   din         Array of input object(s)
```

File: horace_core/sqw/@DnDBase/compact.m [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+%
+% Output:
+% -------
+%   dout        Output object(s), with length of axes reduced to yield the
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@DnDBase/compact.m:19]:
```suggestion
%   dout        Scalar output object or cell array of output object(s), with length of axes reduced to yield the
```

File: horace_core/sqw/@DnDBase/compact.m [hunk 10]
```diff
@@ -0,0 +1,49 @@
+function dout = compact(din)
+% Squeezes the data range in a dnd object to eliminate empty bins on
+% borders
+%
+% Particularly of use for contracting dnds for plotting
+%
+% If an array of DnDs is provided, this will return a cell array
+% otherwise it will return a DnD scalar
+%
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@DnDBase/compact.m:9]: An equivalent method is provided for the sqw class to compact its dnd sub-objects

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]

**Review by cmarooney-stfc — APPROVED** [review]

**Review by cmarooney-stfc — APPROVED** [review]

**Pull request merged**